src/ShallowTraversal.js

```diff
@@ -110,10 +110,13 @@ export function buildPredicate(selector) {
           return node => nodeHasId(node, selector.slice(1));
 
         case SELECTOR.PROP_TYPE: {
-          const propKey = selector.split(/\[([a-zA-Z-]*?)(=|])/)[1];
+          const propKey = selector.split(/\[([a-z]+[a-z-0-9-]*?)(=|])/i);
+          if (propKey.length === 1) {
+            throw new TypeError(`Enzyme::Invalid attribute selector ('${selector}')`);
+          }
           const propValue = selector.split(/=(.*?)]/)[1];
```

Comment:
`const [, propValue] = selector.split(/=(.*?)]/);` (i know this is existing code)